crates/ol/stf/src/update.rs

```diff
@@ -0,0 +1,124 @@
+use strata_acct_types::{AccountId, BitcoinAmount, MsgPayload};
+use strata_ledger_types::{
+    AccountTypeState, Coin, IAccountState, IGlobalState, ISnarkAccountState, StateAccessor,
+};
+use strata_ol_chain_types_new::OLLog;
+
+use crate::{
+    error::{StfError, StfResult},
+    handlers::{
+        get_system_msg_handler, get_system_transfer_handler, handle_snark_msg,
+        handle_snark_transfer,
+    },
+    verification::VerifiedUpdate,
+};
+
+pub(crate) fn apply_update_outputs<'a, S: StateAccessor>(
+    state_accessor: &mut S,
+    sender: AccountId,
+    sender_state: &mut S::AccountState,
+    verified_update: VerifiedUpdate<'a>,
+) -> StfResult<Vec<OLLog>> {
+    let outputs = verified_update.operation().outputs();
+    let transfers = outputs.transfers();
+    let messages = outputs.messages();
+    let mut logs = Vec::new();
+
+    // Process transfers
+    for transfer in transfers {
+        let xfer_logs = send_transfer(state_accessor, sender, transfer.dest(), transfer.value())?;
+        logs.extend(xfer_logs);
+    }
+
+    // Process messages
+    for msg in messages {
+        let payload = msg.payload();
+        let msg_logs = send_message(state_accessor, sender, msg.dest(), payload)?;
+        logs.extend(msg_logs);
+    }
+
+    // Update balance
+    let total_sent = outputs
+        .total_output_value()
+        .ok_or(StfError::BitcoinAmountOverflow)?;
+
+    let _coins = sender_state.take_balance(total_sent);
+    // TODO: do something with coins
+
+    // Update account type state
+    // TODO: think about where it makes sense to keep this logic.
+    match sender_state.get_type_state_mut()? {
+        AccountTypeState::Empty => {}
+        AccountTypeState::Snark(st) => {
+            let operation = verified_update.operation();
+            let new_state = operation.new_state();
+            st.set_proof_state_directly(
+                new_state.inner_state(),
+                new_state.next_inbox_msg_idx(),
+                operation.seq_no(),
+            );
+        }
+    };
```

Comment:
This is mixing up types improperly.  The account's type should never be something we have to match on if we're talking about updates since updates are *always* for snark account types.

Also, you are correct, this is not an *output*, so it doesn't make sense to be here.